shared-bindings/pixelbuf/PixelBuf.c

```diff
@@ -0,0 +1,506 @@
+        mp_get_buffer_raise(args[ARG_rawbuf].u_obj, &rawbufinfo, MP_BUFFER_READ | MP_BUFFER_WRITE);
+        if (rawbufinfo.len != bufinfo.len) {
+            mp_raise_ValueError("rawbuf is not the same size as buf");
+        }
+    }
+    
+    if (size > bufinfo.len) {
+        mp_raise_ValueError_varg("buf is too small.  Need at least %d bytes.", size);
+    }
+
+    pixelbuf_pixelbuf_obj_t *self = m_new_obj(pixelbuf_pixelbuf_obj_t);
+
+    self->base.type = &pixelbuf_pixelbuf_type;
+    self->pixels = args[ARG_size].u_int;
+    self->bpp = bpp;
+    self->bytes = size;
+    self->byteorder = args[ARG_byteorder].u_int;
+    self->bytearray = args[ARG_buf].u_obj;
+    self->buf = bufinfo.buf;
+    self->rawbuf = two_buffers ? rawbufinfo.buf : NULL;
+    self->two_buffers = two_buffers;
+
+    if (args[ARG_brightness].u_obj == mp_const_none) {
+        self->brightness = 1.0;
+    } else {
+        self->brightness = mp_obj_get_float(args[ARG_brightness].u_obj);
+    }
+
+    if (two_buffers) {
+        self->rawbytearray = args[ARG_rawbuf].u_obj;
+    }
+    
+    return MP_OBJ_FROM_PTR(self);
+}
+
+
+
+STATIC mp_obj_t pixelbuf_pixelbuf_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value) {
+    if (value == MP_OBJ_NULL) {
+        // delete item
+        // slice deletion
+        return MP_OBJ_NULL; // op not supported
+    } else {
+        pixelbuf_pixelbuf_obj_t *self = MP_OBJ_TO_PTR(self_in);
+        if (0) {
+#if MICROPY_PY_BUILTINS_SLICE
+        } else if (MP_OBJ_IS_TYPE(index_in, &mp_type_slice)) {
+            mp_bound_slice_t slice;
+            mp_buffer_info_t bufinfo;
+            if (!mp_seq_get_fast_slice_indexes(self->bytes, index_in, &slice)) {
+                mp_raise_NotImplementedError("only slices with step=1 (aka None) are supported");
+            }
+            if ((slice.stop * self->bpp) > self->bytes) 
+                mp_raise_IndexError("Range beyond bounds of pixel buffer");
+
+            if (value != MP_OBJ_SENTINEL) {
+                #if MICROPY_PY_ARRAY_SLICE_ASSIGN
+                size_t dst_len = slice.stop - slice.start;
+                size_t dst_size = dst_len * self->bpp;
+                uint8_t* src_items;
+                uint8_t *destbuf = NULL, *adjustedbuf = NULL;
+                if (self->two_buffers) {
+                    destbuf = self->rawbuf;
+                    adjustedbuf = self->buf;
+                } else {
+                    destbuf = self->buf;
+                }
+                if (MP_OBJ_IS_TYPE(value, &mp_type_array) ||
+                        MP_OBJ_IS_TYPE(value, &mp_type_bytearray) ||
+                        MP_OBJ_IS_TYPE(value, &mp_type_memoryview) ||
+                        MP_OBJ_IS_TYPE(value, &mp_type_bytes)) {
+                    mp_get_buffer_raise(value, &bufinfo, MP_BUFFER_READ);
+                    if (1 != mp_binary_get_size('@', bufinfo.typecode, NULL)) {
+                        mp_raise_ValueError("Array values should be single bytes.");
```

Comment:
Why? Wouldn't it make more sense as an array of 32 bit values, one for each color? That or a list of tuples.